docs/contributing.rst

```diff
@@ -147,34 +153,68 @@ Some tips for reviewing
 
 
 .. code-block:: python
-
     def znl_mean(ar):
         return ar.mean('xt_ocean')
-
-has a few issues. First, the names don't read English. The method does not have any documentation nor is self-explanatory. Further, the method assumes that ``xt_ocean`` is a coordinate of the data array.
+has a few issues. First, the names aren't easily understood. The method does not have any documentation, nor is it self-explanatory. Furthermore, the method assumes that ``xt_ocean`` is a coordinate of the data array; hard coding dimension names is fragile to future changes.
 
 A much better version, free from all the cons mentioned above, is:
 
 .. code-block:: python
-
     def zonal_mean(dataarray):
         '''
         Returns the (numerical) zonal mean of `dataarray`, i.e., its mean along latitude circles.
-
             Parameters:
                     dataarray (xarray.dataarray): An xarray dataarray
-
             Returns:
                     xarray.dataarray: The (numerical) zonal mean of `dataarray`
         '''
-
         return dataarray.cf.mean('longitude')
-     
-
 * Ensure that the notebook runs! To do that:
 
   - Clone the repository or the fork that the PR was made from;
   - Checkout the appropriate branch;
   - Ensure that the notebook runs when a **new** kernel is launched. Ensure that all cells run in sequential order, and that all cell outputs are evaluated.
 
 .. _existing Pull Requests: https://github.com/COSIMA/cosima-recipes/pulls
+
+
+To clone a pull request locally
++++++++++++++++++++++++++++++++
+
+If you want to test pull requests locally (i.e., to compile or run the code),
+you will need to download the pull request branch. You can do this either by cloning the branch from the pull request.
+
+In this context "locally" means somewhere you can run the code - this is probably on Gadi, but may also be on a local machine.
+
+If you are using ssh keys for command line authentication:
+
+::
+
+    git clone -b «THEIR_DEVELOPMENT_BRANCHNAME» git@github.com:«THEIR_GITHUB_USERNAME»/cosima-recipes.git
```

Comment:
I personally never used `-b` ... I just clone the fork...
So I'm confused by this instruction. Or I may not have being doing it correctly?

Reply:
This command only clones the branch of interest. You can definitely clone their whole fork, but I find I get confused if they have lots of unhelpfully/similarly named branches. 

It's a trade off in terms of complexity - this command is more complex than merely cloning the whole thing, but makes the next steps easier.

Reply:
> only clones the branch of interest

It actually still clones the entire repository, just saves you having to `checkout` the branch in another step. There are things like `--single-branch`, but sometimes they can cause more trouble than they're worth!

Reply:
Thanks @angus-g! Good to know.

Reply:
I also feel that this would work for users that are more familiar with git but might confuse less experienced users.

@adele-morrison what's your take?